jwst/exp_to_source/exp_to_source.py

```diff
@@ -33,15 +33,24 @@ def exp_to_source(inputs):
         The key is the ID of each source, i.e. ``source_id``.
     """
     result = DefaultOrderedDict(MultiExposureModel)
+
     for exposure in inputs:
-        log.info('Reorganizing data from exposure {}'.format(exposure.meta.filename))
+        log.info(f'Reorganizing data from exposure {exposure.meta.filename}')
+
         for slit in exposure.slits:
-            log.debug('Copying source {}'.format(slit.source_id))
-            result[str(slit.source_id)].exposures.append(slit)
-            merge_tree(
-                result[str(slit.source_id)].exposures[-1].meta.instance,
-                exposure.meta.instance
-            )
+            log.debug(f'Copying source {slit.source_id}')
+            result_slit = result[str(slit.source_id)]
+            result_slit.exposures.append(slit)
+            merge_tree(result_slit.exposures[-1].meta.instance, exposure.meta.instance)
+
+            if result_slit.meta.instrument.name is None:
```

Comment:
Question: This check is to ensure that the update happens only once?

Quesrtion: The next check on whether inputs is a `DataModel`: The input is always a list, no? What situation is this true?

Reply:
Not actually sure; essentially I just pulled these specific changes from the other PR as a starting point like we said. I'll look into it

Reply:
I think that the first condition checks to see whether additional header info should be added. 

The second might be to handle the case where the input isn't a list, but a MultiExposureModel or other model for a single exposure?

Reply:
Go figure and explains why I'm questioning them. The first check makes sense, but why a non-list would ever be provided here, or allowed, I'm not sure of. I would remove that and ensure at least unit tests pass. Otherwise, should be good.